app/controllers/topbar_sweeper.rb

```diff
@@ -18,7 +18,8 @@ def after_select(record)
   end
 
   def self.fragment_name(id = User.current.id)
-    "tabs_and_title_records-#{id}"
+    plugin_list_hash = Foreman::Plugin.all.map(&:id).sort.hash.abs.to_s(16)
+    "tabs_and_title_records-#{plugin_list_hash}-#{id}"
```

Comment:
Exactly what @domcleal said, what I plan to do is actually the cleanest way IMHO. I just need to base it off menu names instead of plugin names.